bundles/org.openhab.binding.teslapowerwall/src/main/java/org/openhab/binding/teslapowerwall/internal/TeslaPowerwallWebTargets.java

```diff
@@ -0,0 +1,138 @@
+        getBatterySOEUri = baseUri + "api/system_status/soe";
+        getGridStatusUri = baseUri + "api/system_status/grid_status";
+        getSystemStatusUri = baseUri + "api/system_status";
+        getMeterAggregatesUri = baseUri + "api/meters/aggregates";
+        getTokenUri = baseUri + "api/login/Basic";
+        getOperationUri = baseUri + "api/operation";
+    }
+
+    public BatterySOE getBatterySOE(String email, String password) throws TeslaPowerwallCommunicationException {
+        String response = invoke(getBatterySOEUri, email, password);
+        logger.trace("getBatterySOE response = {}", response);
+        return BatterySOE.parse(response);
+    }
+
+    public GridStatus getGridStatus(String email, String password) throws TeslaPowerwallCommunicationException {
+        String response = invoke(getGridStatusUri, email, password);
+        logger.trace("getGridStatus response = {}", response);
+        return GridStatus.parse(response);
+    }
+
+    public SystemStatus getSystemStatus(String email, String password) throws TeslaPowerwallCommunicationException {
+        String response = invoke(getSystemStatusUri, email, password);
+        logger.trace("getSystemStatus response = {}", response);
+        return SystemStatus.parse(response);
+    }
+
+    public MeterAggregates getMeterAggregates(String email, String password)
+            throws TeslaPowerwallCommunicationException {
+        String response = invoke(getMeterAggregatesUri, email, password);
+        logger.trace("getMeterAggregates response = {}", response);
+        return MeterAggregates.parse(response);
+    }
+
+    public Operations getOperations(String email, String password) throws TeslaPowerwallCommunicationException {
+        String response = invoke(getOperationUri, email, password);
+        logger.trace("getOperations response = {}", response);
+        return Operations.parse(response);
+    }
+
+    public String getToken(String email, String password) throws TeslaPowerwallCommunicationException {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty("username", "customer");
+        jsonObject.addProperty("password", password);
+        jsonObject.addProperty("email", email);
+        jsonObject.addProperty("force_sm_off", false);
+        logger.debug("logonjson = {}", jsonObject.toString());
+        String response = invoke(getTokenUri, "POST", "Content-Type", "application/json", jsonObject.toString());
+        JsonObject jsonResponse = JsonParser.parseString(response).getAsJsonObject();
+        String token = jsonResponse.get("token").getAsString();
+        logger.debug("Token: {}", token);
+        return token;
+    }
+
+    private String invoke(String uri, String email, String password) throws TeslaPowerwallCommunicationException {
+        if (token.isEmpty()) {
+            token = getToken(email, password);
+        }
+        return invoke(uri, "GET", "Authorization", "Bearer " + token, "");
```

Comment:
Minor, but prefer that the request method is an enum from jetty lib and not a string.

Reply:
Sorry, I don't really understand  this one.

Reply:
I mean this enum https://javadoc.jetty.org/jetty-12/org/eclipse/jetty/http/HttpMethod.html

Better to use this instead of magic strings

Reply:
OK latest commit (I think) does what you mean here. ie replaces "GET". I also changed the call above that used "POST"